### PR TITLE
expr/builtin_fncall: implement PI

### PR DESCRIPTION
**Pull request by sweetIan**
## What have you changed? (mandatory)

This PR implements the scalar function PI

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- New feature (non-breaking change which adds functionality)

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Refer to a related PR or issue link (optional)

#3275 

The related source in tidb: [https://github.com/pingcap/tidb/blob/55d8eff75859814114f9c6aa3efe2e3f69f7981f/expression/builtin_math.go#L1614](url)

### CODE REVIEW

**Comment by breeswish** [comment]
Thanks a lot! @AndreMouche PTAL

**Comment by breeswish** [comment]
@AndreMouche what do think about DuChuan's thought, which is, for these constant functions or randomize functions, they don't need to be pushed down to TiKV?

**Comment by zz-jason** [comment]
It should be a constant if we pushed it to TiKV:
```sql
TiDB(localhost:4000) > desc select * from t where a > pi();
+---------------------+----------+------+------------------------------------------------------------+
| id                  | count    | task | operator info                                              |
+---------------------+----------+------+------------------------------------------------------------+
| TableReader_7       | 3333.33  | root | data:Selection_6                                           |
| └─Selection_6       | 3333.33  | cop  | gt(test.t.a, 3.141592653589793)                            |
|   └─TableScan_5     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo |
+---------------------+----------+------+------------------------------------------------------------+
3 rows in set (0.00 sec)
```

**Comment by zz-jason** [comment]
@breeswish @spongedu For those functions, we still need to implement them in TiKV:
- give a chance to the `project` operator to be parallel executed in the coprocessor.
- give a chance to the `filter` operator to filter more recored out.

**Comment by hicqu** [comment]
LGTM.

File: src/coprocessor/dag/expr/builtin_math.rs [hunk 1]
```diff
@@ -108,6 +108,10 @@ impl ScalarFunc {
         self.children[0].eval_int(ctx, row)
     }
 
+    #[inline]
+    pub fn pi(&self, _ctx: &mut EvalContext, _row: &[Datum]) -> Result<Option<f64>> {
+        Ok(Some(f64::consts::PI))
+
```
Comment (breeswish) [inline src/coprocessor/dag/expr/builtin_math.rs]: Missing `}`

**Comment by AndreMouche** [comment]
@breeswish @hicqu Please approve it.

**Pull request merged**